### ISSUE TITLE
Binary Cross Entropy的理解

### ISSUE THREAD

**Issue by rentainhe** [rentainhe/read-papers#13]
## 参考
- 译文：https://zhuanlan.zhihu.com/p/89391305
- 原文：https://towardsdatascience.com/understanding-binary-cross-entropy-log-loss-a-visual-explanation-a3ac6025181a

## 问题假设
假设目前有一个二分类问题，有十个点，我们为这十个点分配红色和绿色的标签
![image](https://user-images.githubusercontent.com/48727989/134772418-7d412b9b-426e-4da7-8fdb-45bb13f458e7.png)
![image](https://user-images.githubusercontent.com/48727989/134772419-6dd3d6b8-0fc6-4401-baf3-f361589e0f44.png)
我们现在分类的问题转换为：**给一个值，预测其为红色或者绿色**

## 损失函数：Binary Cross-Entropy / Log Loss
- 二值交叉熵损失函数或者二值对数损失函数
![image](https://user-images.githubusercontent.com/48727989/134772488-fbd7bc31-abd5-4882-b860-770c15153d1e.png)

### 根据正负样本，分开所有点
![image](https://user-images.githubusercontent.com/48727989/134772530-290c07e9-c759-41aa-9fc1-64589bf102ec.png)

### 根据样本，拟合一条Logistic Regression回归曲线来进行二分类
![image](https://user-images.githubusercontent.com/48727989/134772538-d3556224-6308-47c6-a32d-e7cc5d9f3b2f.png)

### 对于所有正类，我们分类器给出的预测概率为（相当于模型输出）
![image](https://user-images.githubusercontent.com/48727989/134772589-aca8204c-b3da-41e0-a192-87615c7556f2.png)

### 对于所有负类，我们分类器给出的预测概率为（相当于模型输出）
![image](https://user-images.githubusercontent.com/48727989/134772596-a20c23fe-b1e7-494d-8efb-6ec5be981922.png)

### 放在一起得到的结果如下
![image](https://user-images.githubusercontent.com/48727989/134772639-860f44ff-9443-4f3d-8b99-d08bd12e5e50.png)
这个条形图表示，每个点与自己**真实类别**对应的概率

### 使用二值交叉熵/对数损失来评估这个性能（相当于求损失函数）
- 此时我们仅仅需要知道所有点预测的概率，来计算这个二值交叉熵损失函数
![image](https://user-images.githubusercontent.com/48727989/134772698-d17fa8b5-e5d3-4306-b525-823188c812d8.png)
![image](https://user-images.githubusercontent.com/48727989/134772702-f78df25b-7304-454e-8839-ffda71423de0.png)
- 我们需要对错误的预测进行惩罚，如果预测出的值，越接近真实标签，那么Loss就应该尽可能越小，离真实标签越远，那么Loss就应该尽可能越大

### 此时采用预测出为真实值的概率的负对数很合适
- 负对数是因为0到1之间的Log为负，我们需要再取一次负，才能得到一个正值的Loss
![image](https://user-images.githubusercontent.com/48727989/134773076-55358d4e-8914-4ac6-a612-2e866715f890.png)

- 我们最终得到的Loss如下，其中可以看到与真实标签越接近，Loss越小
![image](https://user-images.githubusercontent.com/48727989/134773105-d0828d4f-d01d-4df9-8f26-d630b0c99d2b.png)
